### PR TITLE
feat: add simplified channel parameter

### PR DESCRIPTION
**Pull request by jonaslagoni**
**Description**
This PR adds the support for the simplified channel parameters by utilizing a custom implementation of Schema that just looks in the parent parameter object for the values instead of under a `.schema` property that no longer exist for v3.

**Related issue(s)**
Fixes https://github.com/asyncapi/parser-js/issues/815

### CODE REVIEW

**Comment by smoya** [comment]
Would you consider throwing warning log lines for those methods that users are unexpected to use?

**Comment by jonaslagoni** [comment]
@smoya I considered it, but I could not justify it with any reason. Cause it's not like the functions are deprecated or anything 🤷 They just wont return anything if paired with v3 documents, but will with v2.

So not sure who the logging is for?

**Comment by magicmatatjahu** [comment]
@jonaslagoni I don't know if it was discussed but you can reuse v3.SchemaModel but passing as `json` object the serialized schema object from new parameter shape, I mean map fields like type etc:

```
{
  type: "string"
  description: parameter.description
}
```

etc.

Creating new object for this is not a good idea.

**Comment by jonaslagoni** [comment]
@magicmatatjahu you right, thats way cleaner, let me refactor it.

**Comment by jonaslagoni** [comment]
Done 👍

File: src/models/v3/channel-parameter.ts [hunk 1]
```diff
@@ -7,7 +7,7 @@ import type { SchemaInterface } from '../schema';
 import type { ExtensionsInterface } from '../extensions';
 
 import type { v3 } from '../../spec-types';
-import { ParameterSchema } from './channel-parameter-schema';
+import { Schema } from './schema';
```
Comment (magicmatatjahu) [inline src/models/v3/channel-parameter.ts:10]: Import in wrong place 😄

**Review by magicmatatjahu — COMMENTED** [review]
Only one comment, rest is good :)

File: src/models/v3/channel-parameter.ts (hunk 1 above)
Comment (jonaslagoni) [reply src/models/v3/channel-parameter.ts]: Done 👍

File: src/models/v3/channel-parameter.ts [hunk 2]
```diff
@@ -15,12 +13,21 @@ export class ChannelParameter extends BaseModel<v3.ParameterObject, { id: string
...
+    );
   }
 
   schema(): SchemaInterface | undefined {
-    if (!this._json.schema) return undefined;
-    return this.createModel(Schema, this._json.schema, { pointer: `${this._meta.pointer}/schema` });
+    if (!this.hasSchema()) return undefined;
+    return this.createModel(Schema, {
+      type: 'string', 
+      enum: this._json.enum,
```
Comment (magicmatatjahu) [inline src/models/v3/channel-parameter.ts:27]:
```suggestion
      description: this._json.description,
      enum: this._json.enum, 
```

**Review by magicmatatjahu — CHANGES_REQUESTED** [review]

**Comment by jonaslagoni** [comment]
@magicmatatjahu do you have a bit of OCD by any chance? 😄

**Review by magicmatatjahu — APPROVED** [review]

**Pull request merged**